oauth.rst

```diff
@@ -1,131 +1,129 @@
+The response from this call has the following
+-  the ``id`` of this user in Dock, which you should store for this user in your system. When authenticating this allows you to compare this id to the ones stored with you, if you find a match for a user then that is the user that has already logged in using DOCK in your application.
+- ETH address of the contract between the Client and the User.
 
-- ``access_token``: will be used by the client to sign requests on behalf of the user
-- ``token_type=bearer``: this is by default
+This is the end of the OAuth flow.
 
-4) Using an Access Token to get User Data
------------------------------------------
-The Client now has an ``access_token`` that it can use on a call to:
-``GET https://app.dock.io/api/v1/oauth/user-data`` to get the actual User Data. This call needs to be made from the BE so the traffic is not visible in the browser. The following params are expected in the call:
 
-- ``client_id``: given to the developer when registering the Client application.
-- ``client_secret``: given to the developer when registering the Client application.
+Appendix: Variable Notes
+==========================
+Following are the notes about some of the variables mentioned above.
+
+State Variable
+--------------
+The standard ``state`` url parameter is returned unmodified back to the Client. Clients are encouraged to use it to prevent CSRF attacks. A good state variable could be a self-signed string containing some simple info like:
+
+- Current url to redirect the user to the right page once back
+- User id
+
+By signing the ``state`` var properly, you will be able to verify its signature and contents. If a CSRF attack took place, the signature will be broken and you should abort that authentication flow.
+
+Redirect URIs
+-------------
+Only HTTPS addresses are accepted as Redirect URIs. This means that native apps are not supported for now. **Exact match** is used during the authentication flow, and it is forbidden to provide URLs containing anything after the fragment identifier.
+
+Scopes
+------
+A ``scope`` is a way to limit a 3rd party app's access to a user's data. There are 2 choices.
 
-Additionally, the call is expected to contain a header like ``Authorization: Bearer <access_token>``
+Basic Scope (``basic``): This scope will only contain the DOCK user id & ETH address of the contract between the Client and the User. The Client can pass this address and ask the ``dock-gateway`` to fetch and decrypt the user data, and in later versions use this address to go and interact with the contract directly in the Ethereum network.
 
-The response from this call is whatever user data the scope gives access to. Right now, with the "basic" scope this data is first- and last name, email, avatar and dock user id. In the future this may change.
+Full Scope (``full``): This scope will share a lot more details about the user with the partner. The complete list is specified in the next Appendix.
```

Comment:
The complete list is no longer specified in the next appendix now that we moved it.